site/cds_rdm/oidc.py

```diff
@@ -3,21 +3,43 @@
 from invenio_userprofiles.forms import confirm_register_form_preferences_factory
 from flask import current_app
 from werkzeug.local import LocalProxy
+from invenio_oauthclient import current_oauthclient
+from invenio_oauthclient.contrib.keycloak.handlers import get_user_info
 
 _security = LocalProxy(lambda: current_app.extensions["security"])
 
 
 def confirm_registration_form(*args, **kwargs):
     Form = confirm_register_form_preferences_factory(_security.confirm_register_form)
+
     class _Form(Form):
         password = None
         recaptcha = None
         submit = None  # defined in the template
+
     return _Form(*args, **kwargs)
 
 
-def cern_info_serializer(remote, resp, token_user_info, user_info):
+def cern_group_serializer(remote, groups, **kwargs):
+    """Serialize the groups response object."""
+    serialized_groups = []
+    # E-groups do have unique names and this name cannot be updated, therefore the name can act as an ID for invenio
+    for group_name in groups:
```

Comment:
why does this have a different format than https://github.com/CERNDocumentServer/cds-rdm/pull/74/commits/932157820d40572ccb53caeffacb58b0d32f1be6#diff-48d7dfeba7dedee600381f6c5491916254d518942b2c2708f42135df63e3a690R59

Reply:
Good question!

Because the information we get is different, when syncing user e-groups on log in, we only get a list of identifiers (`["zenodo-admins", cds-admins", ...]`), so we set the id with that value and the name with the same value (which in CERN e-groups is always the same). 

The referenced PR you have pasted (group syncing cronjob), does fetch all the groups with all the information, therefore, we will also get the description (and the name in a separate attribue although it's the same as the id), so that we can complete the information of the groups. 

This was agreed with Nico and the auth team.